src/modules/logicModule/LM_PC_Bounties_v1.sol

```diff
@@ -335,23 +349,37 @@ contract LM_PC_Bounties_v1 is ILM_PC_Bounties_v1, ERC20PaymentClientBase_v1 {
         validPayoutAmounts(minimumPayoutAmount, maximumPayoutAmount)
         returns (uint id)
     {
-        // Note ids start at 1.
-        uint bountyId = ++_nextId;
-
-        // Add Bounty id to the list.
-        _bountyList.addId(bountyId);
-
-        Bounty storage b = _bountyRegistry[bountyId];
-
-        b.minimumPayoutAmount = minimumPayoutAmount;
-        b.maximumPayoutAmount = maximumPayoutAmount;
-        b.details = details;
-
-        emit BountyAdded(
-            bountyId, minimumPayoutAmount, maximumPayoutAmount, details
-        );
+        return _addBounty(minimumPayoutAmount, maximumPayoutAmount, details);
+    }
 
-        return bountyId;
+    /// @inheritdoc ILM_PC_Bounties_v1
+    function addBountyBatch(
+        uint[] calldata minimumPayoutAmounts,
+        uint[] calldata maximumPayoutAmounts,
+        bytes[] calldata detailArray
+    )
+        external
+        onlyModuleRole(BOUNTY_ISSUER_ROLE)
+        validArrayLengths(
+            minimumPayoutAmounts.length,
+            maximumPayoutAmounts.length,
+            detailArray.length
+        )
+        returns (uint[] memory ids)
+    {
+        uint arrayLength = minimumPayoutAmounts.length;
+        // create new array to return
+        ids = new uint[](arrayLength);
+
+        for (uint i = 0; i < arrayLength; i++) {
+            // check payout amounts
+            _validPayoutAmounts( //@note should I check first and then add in a different loop?
```

Comment:
Do you mean in terms of when it fails (rather earlier than later)? I don't have a preference, just wanna understand your motives for checking early potentially.

Reply:
I could do 2 Loops for a more separated "modifier" Check
The difference would be less gas wasted in case of a fail state. Thats it.
Then I would just keep it the way it is

Reply:
Yeah we can keep it like this then, good idea nevertheless though!